engine/battle/effect_commands.asm

```diff
@@ -4823,9 +4823,43 @@ UpdateMoveData:
 	push af
 
 	; Write to move selection
+	; Support for duplicate moves: If wCur[Enemy]MoveNum already points to the
+	; current move, preserve it. This allows trainers to have multiple instances
+	; of the same move (e.g., Lance's Dragonite with two Hyper Beams).
```

Comment:
I think the comments below are already more than sufficient to clarify that duplicate move support exists and how. Also Lance's Dragonite does not have "two Hyper Beams"; given that it's still impossible to *teach* duplicate moves (although it's nice if the battle engine doesn't totally break on them), that would be an illegal mon for him to have.
```suggestion
```